packages/conventional-commit-lint/src/conventional-commit-lint.ts

```diff
@@ -38,6 +50,56 @@ export = (app: Probot) => {
         );
         return;
       }
+      // Create a failing check if an attempt is made to submit an invalid config file.
+      try {
+        const configChecker = new ConfigChecker<Configuration>(
+          schema,
+          CONFIGURATION_FILE
+        );
+        const valid = await configChecker.validateConfigChanges(
+          context.octokit,
+          owner,
+          repo,
+          context.payload.pull_request.head.sha,
+          context.payload.pull_request.number
+        );
+        // valid === false indicates that a failed check was created on the PR,
+        // we can return immediately in this case:
+        if (valid === false) {
+          logger.warn(
+            `Invalid config file in PR ${context.payload.pull_request.number} for ${owner}/${repo}`
+          );
+          return;
+        }
+      } catch (e) {
+        const err = e as Error;
+        err.message = `Error validating configuration: ${err.message} ${owner}/${repo}`;
+        logger.warn(err);
+      }
+      // Conventional Commit Lint (unlike most automations) is opt-out, vs.,
+      // opt in. For this reason config is loaded with default values.
+      let config: Configuration | undefined = undefined;
+      try {
+        config = await getConfigWithDefault<Configuration>(
+          context.octokit,
+          owner,
+          repo,
+          CONFIGURATION_FILE,
+          {},
+          {schema: schema}
```

Comment:
I think if I had a config like:

```yaml
enabled: false
foo: bar
```

Then `getConfigWithDefault` will throw an error and `config` would not be set. In this case, the bot would still run the linter. Is this the desired outcome?

Reply:
Oh @chingor13 is right

FYI: The error is [`InvalidConfigurationFormat`](https://github.com/googleapis/repo-automation-bots/blob/4ffc71b49c286ea0322fc76945574d1d82b5fa0d/packages/bot-config-utils/src/bot-config-utils.ts#L438)

Reply:
I dropped the `{schema: schema}`, I think it's sufficient that we validate the configuration as they're added to repos.